lib/base_jobmaster.go

```diff
@@ -63,6 +63,7 @@ type JobMasterImpl interface {
 	OnWorkerOnline(worker WorkerHandle) error
 	OnWorkerOffline(worker WorkerHandle, reason error) error
 	OnWorkerMessage(worker WorkerHandle, topic p2p.Topic, message interface{}) error
+	OnWorkerStatusUpdated(worker WorkerHandle, newStatus *WorkerStatus) error
```

Comment:
is it necessary  to provide this method?

Reply:
The business logic might want to handle changes to the `ext` field. Providing this method seems the most convenient.